packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/test/MrwSerializationTypeProviderTests.cs

```diff
@@ -8,11 +8,13 @@
 using System.Linq;
 using System.Reflection;
 using Microsoft.CodeAnalysis;
-using Microsoft.Generator.CSharp.ClientModel.Providers;
 using Microsoft.Generator.CSharp.Input;
 using Microsoft.Generator.CSharp.Providers;
 using Moq;
 using NUnit.Framework;
+using System.Text.Json;
```

Comment:
We should move this https://github.com/Azure/autorest.csharp/blob/feature/v3/test/AutoRest.TestServer.Tests/ModelReaderWriterValidation/ModelTests.cs#L15 and this https://github.com/Azure/autorest.csharp/blob/feature/v3/test/AutoRest.TestServer.Tests/ModelReaderWriterValidation/ModelJsonTests.cs over to this repo and create a concrete implementation for all of our models here.  This does a series of round trip serialization and validates that the logic we write out is correct.